### PR TITLE
Improved debugging of bad variables in codegen

### PR DESCRIPTION
**Pull request by sjoelund**
No real changes; just improving the C-code so you can see which
variables defines correspond to.

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-17_15-28-09](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4764/).

File: Compiler/Template/CodegenCFunctions.tpl [hunk 1]
```diff
@@ -4098,9 +4098,9 @@ template crefToCStr(ComponentRef cr, Integer ix, Boolean isPre, Boolean isStart)
   case SIMVAR(varKind=DAE_RESIDUAL_VAR())
   case SIMVAR(index=-2)
   then
-    if intEq(ix,0) then (if isPre then "$P$PRE")+crefDefine(cr)
-    else '_<%if isPre then "$P$PRE"%><%crefDefine(cr)%>(<%ix%>)'
-  case var as SIMVAR(index=-1) then error(sourceInfo(), 'crefToCStr got index=-1 for <%variabilityString(varKind)%> <%crefStr(name)%>')
+    if intEq(ix,0) then (if isPre then "$P$PRE")+crefDefine(cr)+' /* <%escapeCComments(crefStrNoUnderscore(cr))%>*/'
+    else '_<%if isPre then "$P$PRE"%><%crefDefine(cr)%>(<%ix%>) /* <%escapeCComments(crefStrNoUnderscore(cr))%>*/'
```
Comment (hkiel) [inline Compiler/Template/CodegenCFunctions.tpl]: these two lines generate nested comments...

Comment (sjoelund) [reply Compiler/Template/CodegenCFunctions.tpl]: Yeah, we seem to use that code is quite weird ways although we shouldn't use #define for variables if possible (*looks over at @wibraun*)... I'll just remove those parts :(

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-10-17_19-36-16](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4765/console).

File: Compiler/Template/CodegenCFunctions.tpl (hunk 1 above)
Comment (wibraun) [reply Compiler/Template/CodegenCFunctions.tpl]: Yeah, agree. Actually I already improved last week the variable creation for jacobian, so now we just need adapt the code generation.

**Pull request closed without merging**